src/mlpack/methods/random_forest/random_forest_regressor.hpp

```diff
@@ -0,0 +1,401 @@
+   * control whether or not the datasetInfo or weights arguments should be
+   * ignored.
+   *
+   * @param data Dataset to train on.
+   * @param datasetInfo Dimension information for the dataset (may be ignored).
+   * @param responses responses for the dataset.
+   * @param weights Weights for each point in the dataset (may be ignored).
+   * @param numTrees Number of trees in the forest.
+   * @param minimumLeafSize Minimum number of points in each leaf node.
+   * @param minimumGainSplit Minimum gain for splitting a decision tree node.
+   * @param maximumDepth Maximum depth for the tree.
+   * @param dimensionSelector Instantiated dimension selection policy.
+   * @param warmStart When set to `true`, it fits new trees and add them to the
+   *     previous forest else a new forest is trained from scratch.
+   * @tparam UseWeights Whether or not to use the weights parameter.
+   * @tparam UseDatasetInfo Whether or not to use the datasetInfo parameter.
+   * @tparam MatType The type of data matrix (i.e. arma::mat).
+   * @return The average entropy of all the decision trees trained under forest.
+   */
+  template<bool UseWeights, bool UseDatasetInfo, typename MatType, typename ResponsesType>
+  double Train(const MatType& data,
+               const data::DatasetInfo& datasetInfo,
+               const arma::Row<ResponsesType>& responses,
+               const arma::rowvec& weights,
+               const size_t numTrees,
+               const size_t minimumLeafSize,
+               const double minimumGainSplit,
+               const size_t maximumDepth,
+               DimensionSelectionType& dimensionSelector,
+               const bool warmStart = false);
+
+  //! The trees in the forest.
+  std::vector<DecisionTreeType> trees;
+
+  //! The average gain of the forest.
+  double avgGain;
+};
+
+/**
+ * Convenience typedef for Extra Trees. (Extremely Randomized Trees Forest)
+ *
+ * @code
+ * @article{10.1007/s10994-006-6226-1,
+ *   author = {Geurts, Pierre and Ernst, Damien and Wehenkel, Louis},
+ *   title = {Extremely Randomized Trees},
+ *   year = {2006},
+ *   issue_date = {April 2006},
+ *   publisher = {Kluwer Academic Publishers},
+ *   address = {USA},
+ *   volume = {63},
+ *   number = {1},
+ *   issn = {0885-6125},
+ *   url = {https://doi.org/10.1007/s10994-006-6226-1},
+ *   doi = {10.1007/s10994-006-6226-1},
+ *   journal = {Mach. Learn.},
+ *   month = apr,
+ *   pages = {3–42},
+ *   numpages = {40},
+ * }
+ * @endcode
+ */
+template<typename FitnessFunction = MSEGain,
+         typename DimensionSelectionType = MultipleRandomDimensionSelect,
+         template<typename> class CategoricalSplitType = AllCategoricalSplit>
+using ExtraTreesRegressor = RandomForestRegressor<FitnessFunction,
+                                DimensionSelectionType,
+                                RandomBinaryNumericSplit,
+                                CategoricalSplitType,
+                                false>;
```

Comment:
Ah, ok, if you want to add this, can you add corresponding documentation in `random_forest_regressor.md`?  You can take a look at the way I did it in `random_forest.md` (which was probably updated since you originally wrote the documentation for this, I know this has been sitting for a while...).